armi/reactor/converters/uniformMesh.py

```diff
@@ -675,7 +751,113 @@ def _checkConversion(self):
+        allMeshes = []
+        lastMeshPoint = src.core.findAllAxialMeshPoints([refAssem])[-1]
+        for a in src.core:
+            # Get the mesh points of the assembly, neglecting the first coordinate
+            # (typically zero) and the last (will be fixed).
+            aMesh = src.core.findAllAxialMeshPoints([a])[1:-1]
+            if len(aMesh) == refNumPoints:
+                allMeshes.extend(aMesh)
+        allMeshes = roundAndRemoveDuplicates(allMeshes, places=2)
+
+        # run kmeans with an increasing number of coordinates until one of three
+        # stop conditions are met
+        # 1. The total error goes below the tolerance
+        # 2. The number of mesh points is twice the number in the reference assembly
+        # 3. Maximum number of iterations is exceeded (10 * refNumPoints)
+        stop = False
+        nMesh = refNumPoints
+        maxIters = refNumPoints * 10
+        iters = 0
+        while not stop:
+            meanMesh, error = kmeans(
+                allMeshes, min(nMesh, len(allMeshes)), iter=10, seed=123456789
+            )
+            sortedCoords = sorted(meanMesh)
+            # if any k-means point is within tolerance of a cluster of the original mesh,
+            # replace the original locations with the k-means coordinate
+            allMeshes = reduceAllMeshes(allMeshes, sortedCoords, meshTolerance)
+
+            # check if reduced mesh is too fine before using it
+            meshSpacing = numpy.array(sortedCoords)[1:] - numpy.array(sortedCoords[:-1])
+            if any(meshSpacing < meshTolerance):
+                # the mesh is too fine, keep going
+                nMesh -= 1
+            else:
+                nMesh += 1
+
+                # stop if we meet error criterion or new mesh gets too fine
+                # infinite loop shouldn't happen, but it is theoretically possible with
+                # nMesh += 1 and nMesh -= 1 oscillating
+                # if this happens, just quit and spit out the mesh after maxIters
+                stop = error < tolerance or nMesh > 2 * refNumPoints or iters > maxIters
+
+            iters += 1
+
+        if error > tolerance:
+            warningMsg = (
+                "Uniform mesh converter failed to converge!"
+                + f"error = {error}, tolerance = {tolerance}\n"
+            )
+            if nMesh > 2 * refNumPoints:
+                warningMsg += (
+                    "- Uniform mesh got too fine!"
+                    f"Number of mesh points = {2 * refNumPoints}"
+                )
+            elif iters > maxIters:
+                warningMsg += (
+                    "- Number of iterations for uniform mesh generator exceeded "
+                    f" before convergence! maxIters = {maxIters}"
+                )
+            runLog.warning(warningMsg)
+
+        self._uniformMesh = numpy.array(allMeshes + [lastMeshPoint])
+
+    def _computeAverageAxialMesh(self, meshTolerance=0.2):
```

Comment:
It is pulled from the case settings onto the `UniformMeshGeometryConverter` and passed into these functions.

The only advantage I see of leaving this as an argument to the method instead of just using the `self._meshTolerance` attribute is that you could call this from a script and vary `meshTolerance` for experimenting/testing purposes. That's what I was doing during development of this PR, but I'm not sure it makes sense to leave it in there indefinitely.